es/es-server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1365,32 +1382,45 @@ public long getIfPrimaryTerm() {
     }
 
     public static class GetResult implements Releasable {
-
-        @Nullable
+        private final boolean exists;
+        private final long version;
         private final DocIdAndVersion docIdAndVersion;
         private final Searcher searcher;
 
-        static final GetResult NOT_EXISTS = new GetResult(null, null);
+        public static final GetResult NOT_EXISTS = new GetResult(false, Versions.NOT_FOUND, null, null);
+
+        private GetResult(boolean exists, long version, DocIdAndVersion docIdAndVersion, Searcher searcher) {
```

Comment:
the new arguments `exists` and `version` are only used in tests by the above static `NOT_EXIST` instance. this is again test only code, please remove.